vault/token_store.go

```diff
@@ -1241,28 +1280,52 @@ func (ts *TokenStore) handleCreateCommon(
 		te.TTL = dur
 	}
 
-	// Prevent attempts to create a root token without an actual root token as parent.
-	// This is to thwart privilege escalation by tokens having 'sudo' privileges.
-	if strutil.StrListContains(data.Policies, "root") && !strutil.StrListContains(parent.Policies, "root") {
-		return logical.ErrorResponse("root tokens may not be created without parent token being root"), logical.ErrInvalidRequest
-	}
-
-	// Set the lesser explicit max TTL if defined
-	if role != nil && role.ExplicitMaxTTL != 0 {
-		switch {
-		case te.ExplicitMaxTTL == 0:
-			te.ExplicitMaxTTL = role.ExplicitMaxTTL
-		default:
-			if role.ExplicitMaxTTL < te.ExplicitMaxTTL {
+	// Set the lesser period/explicit max TTL if defined both in arguments and in role
+	if role != nil {
+		if role.ExplicitMaxTTL != 0 {
+			switch {
+			case te.ExplicitMaxTTL == 0:
 				te.ExplicitMaxTTL = role.ExplicitMaxTTL
+			default:
+				if role.ExplicitMaxTTL < te.ExplicitMaxTTL {
+					te.ExplicitMaxTTL = role.ExplicitMaxTTL
+				}
+				resp.AddWarning(fmt.Sprintf("Explicit max TTL specified both during creation call and in role; using the lesser value of %d seconds", int64(te.ExplicitMaxTTL.Seconds())))
+			}
+		}
+		if role.Period != 0 {
+			switch {
+			case te.Period == 0:
+				te.Period = role.Period
+			default:
+				if role.Period < te.Period {
+					te.Period = role.Period
+				}
+				resp.AddWarning(fmt.Sprintf("Period specified both during creation call and in role; using the lesser value of %d seconds", int64(te.Period.Seconds())))
 			}
-			resp.AddWarning(fmt.Sprintf("Explicit max TTL specified both during creation call and in role; using the lesser value of %d seconds", int64(te.ExplicitMaxTTL.Seconds())))
 		}
 	}
 
 	sysView := ts.System()
 
-	// Run some bounding checks if the explicit max TTL is set
+	if role != nil && role.Period > 0 {
```

Comment:
In case if retaining this block gives better confidence on the code, adding a check in this block for `if te.Period != 0 && role.Period < te.Period`, and setting the lesser value might do the job?